### PR TITLE
[Fixes #316] Fix email

### LINKED ISSUE #316: Email validation doesn't accept uppercase characters
### Current Behavior

The email validation in `BackEndApp/authv1/models.py` uses the following regex `^[a-z0-9]+[\._]?[a-z0-9]+[@]\w+[.]\w{2,3}$`
It doesn't allow upper case characters.

### Expected behavior/code

Should accept uppercase characters.

### ⚙️ Environment

- `Python` version(s): 3.8.5
- `OS`: Ubuntu 20.04

### Possible Solution

Update the regex to accept uppercase characters or else connect the email to lower case before validating using the regex. Using the latter would be better since email addresses should be case-sensitive.

### PR DESCRIPTION
**Pull request by mharyam**
# Pull Request

## What does this PR do?
Fixes #316 

- change email to lower before validating on the  User model 
- change email to lower before saving on the  User model 

## What part does this affect?
- [ ] FrontEnd.
- [x] BackEnd.
- [ ] Documentation.
- [ ] Other. (Please specify below)

## Before submitting
- [x] Was this discussed/approved via a GitHub issue or slack?
- [x] Did you read the [contributor guideline](https://github.com/Auto-DL/Auto-DL/blob/v1-beta/CONTRIBUTING.md)?
- [x] Did you ensure that there aren't any other open [Pull Requests](https://www.github.com/Auto-DL/Generator/pulls) for the same update/change?
- [x] Did you make sure the title is self-explanatory and the description concisely explains the PR?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure the code is clean and docstrings have been added or updated as required?
- [x] Did you make sure the code is linted/formatted locally prior to submission? (using `black` and/or `prettier`)
- [ ] Did you make sure to update the documentation with your changes? (if necessary)

## PR review
Anyone in the community is free to review the PR once the tests have passed.

Thank you for contributing to [AutoDL](https://github.com/Auto-DL/Auto-DL). We look forward to your continued support.

> Also using a template like - `[Fixes #issueNumber] Pr Title` for the title of the PR would be recommended in order to merge it 😄
> 
> P.s. Something like `[Fixes #316] Fix email` would make be amazing xD

Thank you for the review 
I have fixed the curly braces issue 

But it will seem difficult to update my commit message  because I have already pushed it online 

I will take not of the naming for next time 😊

### CODE REVIEW

File: BackEndApp/authv1/models.py [hunk 1]
```diff
@@ -41,9 +41,8 @@ def create(self):
             "password": hashed_password,
             "first_name": self.attributes.get("first_name", ""),
             "last_name": self.attributes.get("last_name", ""),
-            "email": self.attributes.get("email"),
+            "email": self.attributes.get("email").lower(),
             "is_verified": False,
-        }
```
Comment (McTechie) [inline BackEndApp/authv1/models.py:46]: @mharyam I think removing this closing bracket for the dictionary would cause the app to break 😅

**Comment by McTechie** [comment]
Also using a template like - `[Fixes #issueNumber] Pr Title` for the title of the PR would be recommended in order to merge it 😄

P.s. Something like `[Fixes #316] Fix email` would make be amazing xD

**Comment by mharyam** [comment]
> Also using a template like - `[Fixes #issueNumber] Pr Title` for the title of the PR would be recommended in order to merge it 😄
> 
> P.s. Something like `[Fixes #316] Fix email` would make be amazing xD


Thank you for the review
I have fixed the curly braces issue

But it will seem difficult to update my commit message because I have already pushed it online

I will take not of the naming for next time 😊

**Comment by McTechie** [comment]
> Thank you @McTechie for the review
> I have fixed the curly braces issue
> 
> But it will seem difficult to update my commit message because I have already pushed it online
> 
> I will take note of the naming for next time 😊

Ahh.. not the commit message, the title of your **Pull Request** 😅 

Try changing the title of this Pull Request from **`Fix email`** to something like **`[Fixes #316] Fix email`** 😄

**Comment by mharyam** [comment]
> > Thank you @McTechie for the review
> > I have fixed the curly braces issue
> > But it will seem difficult to update my commit message because I have already pushed it online
> > I will take note of the naming for next time 😊
> 
> Ahh.. not the commit message, the title of your **Pull Request** 😅
> 
> Try changing the title of this Pull Request from **`Fix email`** to something like **`[Fixes #316] Fix email`** 😄


Thank you 😅
It has been fixed

**Review by RusherRG — APPROVED** [review]

**Pull request merged**